pkg/api/regularApi.go

```diff
@@ -70,27 +71,47 @@ func GetTwinIdentity(opts *GetIdentityOpts) (register.RegisteredIdentity, error)
 
 // UserDelegatesAuthenticationToAgent User delegates authentication to agent.
 func UserDelegatesAuthenticationToAgent(resolverClient register.ResolverClient, userIdentity register.RegisteredIdentity, agentIdentity register.RegisteredIdentity, delegationName string) error {
-	return advancedapi.DelegateAuthentication(resolverClient, userIdentity.KeyPair(), userIdentity.Did(), agentIdentity.KeyPair(), agentIdentity.Did(), delegationName)
+	opts := advancedapi.DelegationOpts{
+		ResolverClient: resolverClient,
+		DelegatingKeyPair: userIdentity.KeyPair(),
+		DelegatingDid: userIdentity.Did(),
+		DelegatingDocument: nil,
```

Comment:
Is my understanding is correct?
1. Regular (e.g. `CreateTwinIdentity` + `TwinDelegatesControlToAgent`) & High-level (e.g. `CreateTwinWithControlDelegation`) APIs make the same number of `/register` & `/discover` calls as before
2. Advanced API (e.g. `DelegateControl`) allows you to specify `DelegatingDocument` & `SubjectDocument` to avoid repeated lookups, but you [still have to perform most of these steps (boomer)](https://github.com/Iotic-Labs/iotic-host/blob/29d8cadae4ff33419e0630c84690ba3d6d3ffcf6/boomer/tests/common/id_manager.go#L143) to avoid the document (twin) being registered first. So in this example, one would save 2x `/discover` (for agent & twin), but still have to perform 2x `/register` twice.

The README only points to [docs.iotics](https://docs.iotics.com/docs/create-decentralized-identity-documents) - which has (incomplete?) examples for Python.
So my question is: Should we have an example to show how to combine twin-create-delegate (to only have one `/register` call)? Or is it only important to us internally? (But: If we're claiming a speed of X - then presumably we have to be able to show the user how to achieve it.)

Maybe I've misunderstood - what would the boomer `createTwinWithControlDelegation` function look like now?

Reply:
I've raised a PR to update boomer. https://github.com/Iotic-Labs/iotic-host/pull/144 

I agree we should have examples of using the iotics-identity-go on docs.iotics. Perhaps a new issue.

Reply:
Placeholder: https://ioticlabs.atlassian.net/browse/EI-301